src/logical/program_analysis/analysis.rs

```diff
@@ -533,7 +541,11 @@ impl Program {
         }
 
         for (rule, analysis) in self.rules().iter().zip(analyses.iter()) {
-            for filter in rule.filters() {
+            for filter in rule
+                .positive_filters()
+                .iter()
+                .chain(rule.negative_filters())
```

Comment:
Again, would be useful to have `Rule::filters` which does the chaining transparently.